### DISCUSSION TITLE
leafn_to_litter with biofuel harvest: Potential imbalance?

### DISCUSSION THREAD

**Discussion by samsrabin** [ESCOMP/CTSM#2146] [Q&A]
It looks like there are three ways that `leafn_to_litter` can be calculated in `CNOffsetLitterfall()`.

If `CNratio_floating` is true:
https://github.com/ESCOMP/CTSM/blob/1e2e2c35d9568c94eac6cb606f37c18294158682/src/biogeochem/CNPhenologyMod.F90#L3232-L3238

If `CNratio_floating` is false and `use_fun` is true:
https://github.com/ESCOMP/CTSM/blob/1e2e2c35d9568c94eac6cb606f37c18294158682/src/biogeochem/CNPhenologyMod.F90#L3188-L3189

If `CNratio_floating` is false and `use_fun` is false:
https://github.com/ESCOMP/CTSM/blob/1e2e2c35d9568c94eac6cb606f37c18294158682/src/biogeochem/CNPhenologyMod.F90#L3212

I'm wondering if it's possible for there to be a discrepancy here with how much N is removed during biofuel harvest, which happens here (before `leafn_to_litter` is calculated):
https://github.com/ESCOMP/CTSM/blob/fdbb234a1d457a5c0fe8db69c30410fe52cde695/src/biogeochem/CNPhenologyMod.F90#L3126

Note that `leafn` is never reduced by `leafn_to_biofueln`, which I think would cause a problem in at least the `CNratio_floating == .true.` case.

Our testing might not be set up to catch such discrepancies and their resulting N imbalances. It's only the Smallville tests that have any biofuels, right?

Pinging @billsacks, as it looks like you did the final testing for the bioenergy crops PR #884.

**Comment by billsacks** [comment]
I haven't gotten my head into everything you are describing here, but it seems like your main point may be this one:

> Note that leafn is never reduced by leafn_to_biofueln

and wanted to confirm that you saw this:

https://github.com/ESCOMP/CTSM/blob/1e2e2c35d9568c94eac6cb606f37c18294158682/src/biogeochem/CNNStateUpdate1Mod.F90#L284

But maybe you're saying that there's an issue even despite that? Let's talk more.

**Comment by samsrabin** [comment]
It actually seems like this is fine. `leafn_to_litter` is in all cases dependent on `leafc_to_litter`, which DOES have biofuel fraction removed. It's also dependent on `leafc`, which like `leafn` is not updated; since biofuel harvest affects leaf C and N equally, then `leafc/leafn` will be the same whether biofuel harvest is removed from both or neither.
